### PR TITLE
Add foxglove_bridge package

### PR DESCRIPTION
**Pull request by defunctzombie**
The foxglove bridge package is a c++ websocket bridge for connecting ROS systems to Foxglove Studio. It is similar in spirit to rosbridge but written in c++ to overcome some performance problems we experienced with rosbridge under large message or high rate message workloads.

https://foxglove.dev/blog/announcing-the-new-foxglove-bridge-for-live-ros-data
https://github.com/foxglove/ros-foxglove-bridge/

Noetic PR: https://github.com/RoboStack/ros-noetic/pull/345

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
I'll add  websocketpp for osx-arm64 and linux-aarch64 as this seems to pass on the other platforms.

**Comment by defunctzombie** [comment]
> I'll add websocketpp for osx-arm64 and linux-aarch64 as this seems to pass on the other platforms.

Was it already available on those other platforms? Or is this something we can improve with our packaging of the bridge package? I never really know whether this should be defined somewhere or we should vendor these types of libraries.

**Comment by Tobias-Fischer** [comment]
RoboStack is based on conda-forge. Thankfully there it is easy to compile for a variety of platforms. Adding linux-aarch64 and osx-arm64 was a matter of adding 5 lines to a config file, the CI does the rest: https://github.com/conda-forge/websocketpp-feedstock/pull/2

So no need to do anything on your side.

You seem to be missing a dependency on `ament_cmake` in your package.xml though. We can fix it manually for now (see commit in a minute), but for a future release consider adding this.

**Comment by defunctzombie** [comment]
> You seem to be missing a dependency on `ament_cmake` in your package.xml though. We can fix it manually for now (see commit in a minute), but for a future release consider adding this.

Cool. We will definitely take care of that.

**Comment by Tobias-Fischer** [comment]
It actually already seems to be in there. It also pulls in ros-humble-ament-cmake and it still can't find the packge. Not sure what's going on ..

**Comment by defunctzombie** [comment]
@Tobias-Fischer 

The linux aarch64 build fails with:
```
CMake Error at CMakeLists.txt:166 (message):
  ROS_VERSION environment variable must be 1 or 2
```

Is this something we need to set in our project? I noticed you attempted to set this with some commits.

**Comment by achim-k** [comment]
> @Tobias-Fischer
> 
> The linux aarch64 build fails with:
> 
> ```
> CMake Error at CMakeLists.txt:166 (message):
>   ROS_VERSION environment variable must be 1 or 2
> ```
> 
> Is this something we need to set in our project? I noticed you attempted to set this with some commits.

Adding a build dependency on `ros_environment` usually fixes that. `foxglove_bridge` has this dependency

**Comment by Tobias-Fischer** [comment]
The same happens for osx-arm64. I'm not sure what's going on ..

**Comment by Tobias-Fischer** [comment]
Do you have any idea why the environment variables are not passed through @wolfv? I set them explicitly in the GitHub actions and still they are not detected by cmake in linux-aarch64 and osx-arm64, but they are in linux-64 and osx-64?!?

File: vinca_win.yaml [hunk 1]
```diff
@@ -49,6 +49,7 @@ packages_select_by_deps:
   - desktop
   - cv_bridge
   - perception
+  - foxglove_bridge
```
Comment (defunctzombie) [inline vinca_win.yaml:52]:
```suggestion
```

**Pull request merged**